app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala

```diff
@@ -175,6 +177,39 @@ class UndertakingController @Inject() (
     )
   }
 
+  def getAddEmailForVerification(status: String = EmailStatus.New.toString): Action[AnyContent] = enrolled.async {
```

Comment:
Not a critique, just a question about why don't  people use

https://www.playframework.com/documentation/2.8.x/ScalaRequestBinders

?

We do it with EORI's as well etc. Go in the controller then blow up with a 500. Things like invalid param format are  really 400. Also less surprise boom.